Sources/JWTKit/X5C/X5CVerifier.swift

```diff
@@ -118,16 +118,16 @@ public struct X5CVerifier: Sendable {
         where Payload: JWTPayload
     {
         // Parse the JWS header to get the header
-        let parser = try JWTParser(token: token)
-        let header = try parser.header(jsonDecoder: jsonDecoder)
+        let parser = try DefaultJWTParser(token: token)
+        let header = try parser.parseHeader(jsonDecoder: jsonDecoder)
 
         // Ensure the algorithm used is ES256, as it's the only supported one (for now)
-        guard let headerAlg = header.alg, headerAlg == "ES256" else {
-            throw JWTError.invalidX5CChain(reason: "Unsupported algorithm: \(header.alg ?? "nil")")
+        guard let headerAlg = header.alg?.asString, headerAlg == "ES256" else {
+            throw JWTError.invalidX5CChain(reason: "Unsupported algorithm: \(String(describing: header.alg))")
         }
 
         // Ensure the x5c header parameter is present and not empty
-        guard let x5c = header.x5c, !x5c.isEmpty else {
+        guard let x5c = try header.x5c?.asArray(of: String.self), !x5c.isEmpty else {
```

Comment:
It might be nice to provide convenience extensions for stuff that's pretty common like this